ext/odbc/php_odbc.c

```diff
@@ -2358,9 +2398,9 @@ void odbc_do_connect(INTERNAL_FUNCTION_PARAMETERS, int persistent)
 			RETURN_FALSE;
 		}
 		ODBCG(num_links)++;
+		zend_hash_update_ptr(&ODBCG(connections), hashed_details.s, Z_ODBC_LINK_P(return_value));
```

Comment:
Accidentally, this line used to be in the wrong place a few lines below so that even persistent connections ended up being in the (non-persistent) connection list. :/

Reply:
I've just realized the bug here too when answering your other comment :slightly_smiling_face: I'll fix this for sure. Although the previous code didn't use connection reuse, but now we should in order not to have to take care of hash collision. I'm uncertain whether this change would cause any problems for people